src/libcore/slice.rs

```diff
@@ -655,6 +655,12 @@ impl<'a, T> Default for &'a [T] {
     fn default() -> &'a [T] { &[] }
 }
 
+#[stable(feature = "mut_slice_default", since = "1.5.0")]
+impl<'a, T> Default for &'a mut [T] {
+    #[stable(feature = "mut_slice_default", since = "1.5.0")]
```

Comment:
@petrochenkov how does that work? You can't possibly use it as a T: Trait in stable code if some of the methods are marked as unstable. If you want to expose some subset as stable, then just have concrete methods for that functionality? The granularity seems useless.